### ISSUE TITLE
Gentle Aqua Raven - Anyone can sweep native token out of any AdvancedDistributor, which prevents paying Connext fees

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2023-06-tokensoft-judging#60]
Gentle Aqua Raven

high

# Anyone can sweep native token out of any AdvancedDistributor, which prevents paying Connext fees

## Summary

`Sweepable.sweepNative` can be called by anyone, allowing anyone to sweep all native token to the owner. Although the current Distributor implementations keep no native token, CrosschainDistributor will need to be changed to keep a supply of native token to pay Connext fees. After doing so, anyone can DOS any attempt to send assets cross-chain.

## Vulnerability Detail

`sweepNative` is not marked `onlyOwner`, even though all other functions in this file are:

https://github.com/sherlock-audit/2023-06-tokensoft/blob/main/contracts/contracts/utilities/Sweepable.sol#L34

```solidity
    function sweepNative() external {
        uint256 amount = address(this).balance;
        (bool success, ) = recipient.call{value: amount}("");
        require(success, "Transfer failed.");
        emit SweepNative(amount);
    }
```

There is another issue where CrosschainDistributor does not pay the Connext fee and therefore cannot send anything cross chain, https://github.com/sherlock-audit/2023-06-tokensoft-jkoppel/issues/8  . The fix will likely be to have the CrosschainDistributor maintain its own supply of native token, as in the official Ping Pong example ( https://docs.connext.network/developers/examples/ping-pong ). Once that fix is in place, anyone will be able to front-run any cross-chain claim with a call to `sweepNative`, thereby preventing anyone else from making a cross-chain claim.

## Impact

After fixing  https://github.com/sherlock-audit/2023-06-tokensoft-jkoppel/issues/8  , anyone can DOS any other cross-chain claim.

## Tool used

Manual Review

## Recommendation

Add the `onlyOwner` modifier to `sweepNative`.
